docs/misc/ConfigDB_state.md

```diff
@@ -0,0 +1,84 @@
+* `NotConfigured` - there is no link for DB in /etc/vmware/vmdkops. No auth config operation is allowed, and the allow_all_access() method will always return True. All requests from vDVS client are auto-approved.
+* `SingleNode` - there is an actual Config DB in /etc/vmware/vmdkops. All authorization system is limited to a single ESX node
+* `MultiNode` - There is  link in /etc/vmware/vmdkops to a Config DB on a shared datastore. Every ESXi node which has similar lin will shared the authorization configuration.
+* `BrokenLink` - Error state. Either the symink point to non-existing file, or the DB file is corrupt. All vDVS client requests will be denined in this mode, and vmdk_ops.py error log will indicate the BrokenLInl state. The only way to recover is to remove the local configuration and re-init the Config DB again
+
+## State transitions
+
+Since there are very few states and the transitions are rather simple, we will outline them here in plan text and skip diagrams.
+
+Currently, there are 2 components setting state:
+`auth_data.AuthenticationDataManager constructor` call (passive setting state for internal usage) and `vmdkopd_admin.py admin CLI commands` (active changing of the state). Going forward, we plan to add another component - vmdk_ops.py service `Config DB periodic discovery`, to assure that internal state reflects changes to Config DB made from other ESXi nodes. Discovery is discussed later in thi document
+
+### From Unknown
+
+On auth_data.AuthenticationDataManager.connect(), we check the actal DB mode change internal state as follows:
+
+* Unknown-> NotConfigured   use "no access control" mode for vmdk_opsd operations
+* Unknown-> SingleNode      open the DB and keep going (backward compat, local DB)
+* Unknown-> MutliNode        open the DB and keep going (shared DB via symlink)
+* Unknown-> BrokenLink      ERROR - throw exception
+
+
+### From NotConfigured
+
+Manual transition from NotConfigured is done by the admin CLI.
+Automatic transition from NotConfigured is done by Discovery (see below)
+
+* NotConfigured -> NotConfigured    Keep going.
+* NotConfigured -> SingleNode       Forced by by creating a new DB in /etc/vmware/vmdkops (AuthenticationDataManager.new_db(). When discovered, we just connect to the DB, log an INFO message and keep going.
+* NotConfigured -> MutliNode         Forced by creating the DB on a shred VMFS volume and creating symlink in /etc/vmware/vmdkops. When discovered, we just connect to the DB, log an INFO message and keep going. (same as in NotConfigured->MutliNode)
+* NotConfigured -> BrokenLink       ERROR - throw exception
+
+### BrokenLink handling
+BrokenLink is an error condition. We do not know what was the admin intent so we fail all Docker volume requests until the situation is resolved by the admin using `config rm --local` command
+
+* BrokenLink    ->  anything forced by admin.
```

Comment:
Do you mean to run "admin config init" when the state is "BrokenLink"? I feel it's tricky/risky to support this transition. We can discuss this when finalizing the state transition conditions in the diagram I just attached.